extensions/amp-next-page/0.2/service.js

```diff
@@ -216,6 +245,50 @@ export class NextPageService {
     this.lastFetchedPage_ = page;
   }
 
+  /**
+   * Sets the title and url of the document to those of
+   * the provided page
+   * @param {?Page=} page
+   */
+  setTitlePage(page = this.initialPage_) {
+    if (!page) {
+      dev().warn('setTitlePage called before next-page-service is built');
+      return;
+    }
+    const {title, url} = page;
+    this.win_.document.title = title;
+    this.history_.replace({title, url});
+  }
+
+  /**
+   * Adds an initial entry in history that sub-pages can
+   * replace when they become visible
+   */
+  initializeHistory() {
+    const {title, url} = this.initialPage_;
+    this.history_.push(undefined /** opt_onPop */, {title, url});
+  }
+
+  /**
+   *
+   * @return {!Page}
+   */
+  createInitialPage() {
+    const doc = this.win_.document;
+    const {title, location} = doc;
+    const {href: url} = location;
+    const image =
+      parseSchemaImage(doc) || parseOgImage(doc) || parseFavicon(doc) || '';
+    return new Page(
+      this,
+      url,
+      title,
+      image,
+      PageState.INSERTED /** optInitState */,
```

Comment:
Update comments. with new variable names. Would also consider switching to a single object param, using destructuring now that you have quite a few arguments,